### ISSUE TITLE
Allow to select currently impossible buildables

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#2543]
Relates to https://github.com/Unvanquished/Unvanquished/issues/364

Currently, when one is playing the builder role, they are unable to "force select" a building that can currently not be done.
This uselessly slows down the building gameplay when you placed a miner and are waiting for it to complete.

My suggestion to improve UX on that is to allow people to select a buildable they don't have the BPs yet, but all other pre-requisites are met (basically: it's unlocked and allowed).

**Comment by slipher** [comment]
Good idea. Your yellow build shader thing will help the user track the situation. Maybe double click/double tap?

**Comment by bmorel** [comment]
> Good idea. Your yellow build shader thing will help the user track the situation. Maybe double click/double tap?

Let me be honest: it's something I have in mind since a long while, it's almost a scheme at this point :D
The yellow shader suffers a huge problem though: it's very hard to distinguish from the other ones. This is a matter of graphics and trivial to change, though.

I'm not sure how exactly to implement this correctly, but if the idea is welcomed, I should be able to setup a server implementing it, so that we can toy with it.

**Comment by DolceTriade** [comment]
This sounds like it needs some thought in how to implement the UX. Done naively (ie, just allowing building anything despite it being disabled/not allowed/etc) would clearly be a detrimental experience. Moving to post-beta as a result. Ofc, if you want to fix earlier than that, feel free to discuss here and an implement an acceptable proposal.
